gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Client.java

```diff
@@ -124,8 +126,8 @@ public Client alias(final Map<String, String> aliases) {
     public ResultSet submit(final Traversal traversal) {
         try {
             return submitAsync(traversal).get();
-        } catch (UnsupportedOperationException uoe) {
-            throw uoe;
+        } catch (RuntimeException re) {
```

Comment:
Thank you for these changes.

1. The top level exception thrown back to the application has changed here from `RuntimeException` to `NoHostAvailableException`. If a user has application level code that is catching the `RuntimeException` and checking the cause as `NoHostAvailableException`, that code will fail with client upgrade. Hence, this is a backward incompatible change since it modifies the exception contract.
For 3.4.x, could we please keep the contract as same (i.e. by wrapping NoHostAvailableException in RuntimeException)?

2. All other changes in nested exceptions look good (we can still have `NoHostAvailableException` with a nice cause).

3. I would vote that we can take this break in exception contract for 3.5.x series since that is still going through some stabilization with respect to other behaviours as well such as null semantics. @spmallette agree?

Overall, I think we do the following:
1. Merge this to master.
2. Create a different version with NoHostAvailableException as a 1st level cause to merge into 3.4 branch.